examples/plot_wlasso.py

```diff
@@ -5,95 +5,116 @@
 """
 
 # Authors: Quentin Bertrand <quentin.bertrand@inria.fr>
 #          Quentin Klopfenstein <quentin.klopfenstein@u-bourgogne.fr>
-#
+#          Kenan Sehic
 # License: BSD (3-clause)
 
-import time
-
 import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from numpy.linalg import norm
 
-from sklearn.datasets import make_regression
+from celer import Lasso, LassoCV
 from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error
+from sklearn.utils import check_random_state
+from scipy.linalg import toeplitz
 
-from celer import Lasso
-
-from sparse_ho.models import wLasso
+from sparse_ho.models import WeightedLasso
 from sparse_ho.criterion import CV
 from sparse_ho.implicit_forward import ImplicitForward
 from sparse_ho.utils import Monitor
 from sparse_ho.ho import grad_search
-from sparse_ho.datasets import get_data
-
-print(__doc__)
 
-dataset = 'rcv1_train'
-# dataset = 'simu'
-
-if dataset != 'simu':
-    X_train, X_val, X_test, y_train, y_val, y_test = get_data(dataset)
-    X_train = X_train[:, :1000]
-    X_test = X_test[:, :1000]
-else:
-    X, y = make_regression(n_samples=100, n_features=100, noise=1)
-    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33)
-    X_train, X_val, y_train, y_val = train_test_split(
-        X_train, y_train, test_size=0.5)
+##############################################################################
+# Dataset creation
+n_samples = 600
+n_features = 600
+rng = check_random_state(0)
+X = rng.multivariate_normal(
+    size=n_samples, mean=np.zeros(n_features),
+    cov=toeplitz(0.5 ** np.arange(n_features)))
+
+
+# Create true regression coefficients of 5 non-zero values
+w_true = np.zeros(n_features)
+size_supp = 5
+idx = rng.choice(
+    X.shape[0], size_supp, replace=False)
+w_true[idx] = (-1) ** np.arange(size_supp)
+noise = rng.randn(n_samples)
+y = X @ w_true
+y += noise / norm(noise) * 0.5 * norm(y)
+
+# Here we split the dataset (X, y) in 3:
+# the regression coefficients will be determined using X_train, y_train
+# the regularization parameter will be calibrated using X_val, y_val
+# the model is then tested on unseen data X_test, y_test
+X_train_val, X_test, y_train_val, y_test = train_test_split(
+    X, y, test_size=0.3, random_state=2)
+X_train, X_val, y_train, y_val = train_test_split(
+    X_train_val, y_train_val, test_size=0.5, random_state=2)
+##############################################################################
 
-n_samples, n_features = X_train.shape
 
-print("Starting path computation...")
-n_samples = len(y_train)
+# Max penalty value
 alpha_max = np.max(np.abs(X_train.T.dot(y_train))) / X_train.shape[0]
-log_alpha0 = np.log(alpha_max / 10)
-
-n_alphas = 10
-p_alphas = np.geomspace(1, 0.0001, n_alphas)
-alphas = alpha_max * p_alphas
-log_alphas = np.log(alphas)
+n_alphas = 30  # number of iter in the line search, ie 30 evals of the gradient
```

Comment:
the naming is confusing, since it is also the number of alphas used in LassOCV below

Reply:
I removed the comment, it was false, the number of line search iterations is set somewhere else